plugins/modules/user_role_workflow_manager.py

```diff
@@ -2434,16 +2442,21 @@ def update_role(self, role_params):
               and the "update_role_api" function. The method logs the received API response at the "DEBUG" level and
               finally returns the response.
         """
-        self.log("Update role with role_info_params: {0}".format(str(role_params)), "DEBUG")
-        response = self.dnac._exec(
-            family="user_and_roles",
-            function="update_role_api",
-            op_modifies=True,
-            params=role_params,
-        )
-        self.log("Received API response from update_role: {0}".format(str(response)), "DEBUG")
+        try:
+            self.log("Update role with role_info_params: {0}".format(str(role_params)), "DEBUG")
+            response = self.dnac._exec(
+                family="user_and_roles",
+                function="update_role_api",
+                op_modifies=True,
+                params=role_params,
+            )
+            self.log("Received API response from update_role: {0}".format(str(response)), "DEBUG")
+            return response
 
-        return response
+        except Exception:
+            error_message = "The catalyst center user '{0}' does not have the necessary permissions to update role through the API.".format(
+                self.payload.get("dnac_username"))
```

Comment:
Can we print the error as well?
    self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")

Reply:
Updated the code accordingly.